source/core/memory.cpp

```diff
@@ -291,36 +292,33 @@ void Memory::deserializeJSON(const Json& json) {
   }
   // TODO::check types!
   // check that sizes match
-  std::size_t byteCount = *byteCountIt;
-  if (_byteCount != byteCount) {
-    std::string expected = std::to_string(_byteCount);
-    std::size_t byteCount = *byteCountIt;
-    std::string recieved = std::to_string(byteCount);
-    throw DeserializationError(
-        "Could not deserialize Memory: ByteCount did not match: expected \"" +
-        expected + "\" to be equal to \"" + recieved + "\"");
+  size_t byteCount = *byteCountIt;
+  if (_byteCount < byteCount) {
+    _byteCount = byteCount;
+    _data = MemoryValue(byteCount * _byteSize);
+    _sizeCallback(_byteCount);
```

Comment:
it'd be also nice if you made a private method that calls the sizeCallback, just like the updateCallback, as that makes this more easily expandable